Add fake client check to CalculateMaxSpeed

M60TM opened the pull request:
```
L 09/03/2023 - 19:55:19: [SM] Exception reported: Entity 22 (22) is invalid
L 09/03/2023 - 19:55:19: [SM] Blaming: tf2attribute_support.smx
L 09/03/2023 - 19:55:19: [SM] Call stack trace:
L 09/03/2023 - 19:55:19: [SM]   [0] GetEntPropEnt
L 09/03/2023 - 19:55:19: [SM]   [1] Line 587, C:\suspawn\mystuff_7040\scripting\tf2attribute_support.sp::OnPlayerCalculateMaxSpeedPost
```
This error also is occurred in SM 1.11. It seems to enough adding index check.

M60TM committed:
Fix Invalid client error on CalculateMaxSpeed

- `L 09/03/2023 - 00:17:06: [SM] Exception reported: Entity 28 (28) is invalid`

nosoop commented:
22 should be within the MaxClients range, so I'd be surprised if your range check works there unless you have a really low player limit set.

M60TM commented:
I got that error log on 33 max player server. It works, but I still don't know how that work. maybe I need more test.

M60TM commented:
After test, I found some interesting points.
1. It is appeared when map change with "bot". (I don't know why it is appeared in server that doesn't have bots.)
2. `IsValidEntity` can't fix this problem.
3. client index check can't fix this problem.
4. I added `IsFakeClient` check, it fixed.

As a result, this error seems to be related with bots.

M60TM committed:
Add Bot check

- keep client index check and added `IsClientInGame` to bot check by using `IsFakeClient`.

nosoop commented on scripting/tf2attribute_support.sp:
```diff
@@ -582,6 +582,10 @@ MRESReturn OnPlayerCanAirDashPre(int client) {
 }
 
 MRESReturn OnPlayerCalculateMaxSpeedPost(int client, Handle hReturn, Handle hParams) {
+	if (client < 1 || client > MaxClients || !IsClientInGame(client) || IsFakeClient(client)) {
```
The `IsFakeClient` check passing here would make it so bots will never be able to use `mult_move_speed_when_active`.

M60TM replied:
You right, I need to find another way.

nosoop reviewed (commented):
`IsValidEntity` failing seems weird.  Maybe there's some other check that can work?

I've usually seen this issue happen only on server shutdown (possibly map change), so fixing it hasn't been that high a priority for me.

If you're seeing it happen elsewhere, reproduction steps would be good; if not, that's fine too &mdash; happy to merge in a working fix here.

M60TM commented:
I can't find a way to solve this. I'll try again later when I get a clearer clue.
